src/segment-loader.js

```diff
@@ -674,12 +683,14 @@ export default videojs.extend(videojs.EventTarget, {
                                                 timelineUpdate);
 
     // the last segment append must have been entirely in the
-    // already buffered time ranges. adjust the timestamp offset to
-    // fetch forward until we find a segment that adds to the
-    // buffered time ranges and improves subsequent media index
-    // calculations.
-//    if (!timelineUpdated) {
-//      this.expired_ -= segment.duration;
-//    }
+    // already buffered time ranges. adjust the timeCorrection
+    // offset to fetch forward until we find a segment that adds
+    // to the buffered time ranges and improves subsequent media
+    // index calculations.
+    if (!timelineUpdated) {
+      this.timeCorrection_ -= segment.duration;
```

Comment:
We talked about this in person. The consensus was that `timeCorrection` has a different lifespan from `expired` and conflating the two variables would lead to heartache.